src/libstore/store-api.hh

```diff
@@ -926,10 +926,18 @@ void copyClosure(
 void removeTempRoots();
 
 
+/**
+ * Attempt to recursively replace derivation outputs with produced paths as
+ * much as possible, but where the derivation resolution doesn't exists, leave
+ * as-is.
+ */
+SingleDerivedPath tryResolveDerivedPath(Store &, const SingleDerivedPath &, Store * evalStore = nullptr);
```

Comment:
Still need to wrap my head around this one, but other than that, I've completed a review of the full diff.

Reply:
I am not even sure if it is needed, tbh.

Reply:
So it is just used by `Installables::toDerivations` (virtual method, multiple implementations). The idea is that we want to try to get the "one up" prior derivation, and fail if it is not built.

But maybe this is bad for the same reason `--derivation` is bad: be explicit, and pass what you mean instead.

Either way, maybe we should just just unconditionally build the thing, and its up to you to pass some sort of `--must-be-already-built` if one doesn't want that.